File: backend/app/schemas/daily_review.py
"""Daily review request and response schemas."""

from __future__ import annotations

from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, Field


class DailyReviewGenerateRequest(BaseModel):
    folder_path: str = Field(..., min_length=1)
    review_date: date
    model: str | None = None


class DailyReviewResponse(BaseModel):
    id: str
    review_date: date
    folder_path: str
    title: str
    content_markdown: str
    source_files: list[dict[str, Any]]
    model: str | None
    provider_id: str | None
    time_created: datetime
    time_updated: datetime

    model_config = {"from_attributes": True}


class DailyReviewListResponse(BaseModel):
    id: str
    review_date: date
    folder_path: str
    title: str
    source_files: list[dict[str, Any]]
    model: str | None
    provider_id: str | None
    time_created: datetime
    time_updated: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/daily-review.ts
export interface DailyReviewSourceFile {
  path: string;
  relative_path: string;
  modified_at: string;
  size: number;
  truncated: boolean;
}

export interface DailyReview {
  id: string;
  review_date: string;
  folder_path: string;
  title: string;
  content_markdown: string;
  source_files: DailyReviewSourceFile[];
  model: string | null;
  provider_id: string | null;
  time_created: string;
  time_updated: string;
}

export type DailyReviewListItem = Omit<DailyReview, "content_markdown">;

export interface DailyReviewGenerateRequest {
  folder_path: string;
  review_date: string;
  model?: string | null;
}
